src/cpp/lib/QtWidgets/QTreeWidgetItem/qtreewidgetitem_wrap.cpp

```diff
@@ -0,0 +1,82 @@
+#include "QtWidgets/qtreewidgetitem/qtreewidgetitem_wrap.h"
+
+#include <iostream>
+
+#include "Extras/Utils/nutils.h"
+#include "QtWidgets/QTreeWidget/qtreewidget_wrap.h"
+#include "core/Component/component_wrap.h"
+
+Napi::FunctionReference QTreeWidgetItemWrap::constructor;
+
+Napi::Object QTreeWidgetItemWrap::init(Napi::Env env, Napi::Object exports) {
+  Napi::HandleScope scope(env);
+  char CLASSNAME[] = "QTreeWidgetItem";
+  Napi::Function func =
+      DefineClass(env, CLASSNAME,
+                  {InstanceMethod("setText", &QTreeWidgetItemWrap::setText),
+
+                   COMPONENT_WRAPPED_METHODS_EXPORT_DEFINE});
+  constructor = Napi::Persistent(func);
+  exports.Set(CLASSNAME, func);
+  return exports;
+}
+
+QTreeWidgetItem* QTreeWidgetItemWrap::getInternalInstance() {
+  return this->instance;
+}
+QTreeWidgetItemWrap::~QTreeWidgetItemWrap() { delete this->instance; }
+
+QTreeWidgetItemWrap::QTreeWidgetItemWrap(const Napi::CallbackInfo& info)
+    : Napi::ObjectWrap<QTreeWidgetItemWrap>(info) {
+  Napi::Env env = info.Env();
+  Napi::HandleScope scope(env);
+  if (info.Length() == 0) {
+    this->instance = new QTreeWidgetItem();
+  } else if (info.Length() >= 1) {
+    int constructorType = info[info.Length() - 1].As<Napi::Number>();
+    if (constructorType == 1) {
+      Napi::Array stringsNapi = info[0].As<Napi::Array>();
+      QList<QString> list;
+      for (int i = 0; i < stringsNapi.Length(); i++) {
+        Napi::Value stringNapi = stringsNapi[i];
+        list.append(stringNapi.As<Napi::String>().Utf8Value().c_str());
+      }
+      QStringList strings = QStringList(list);
+      this->instance = new QTreeWidgetItem(strings);
+    } else if (constructorType == 2) {
+      Napi::Object parentObject = info[0].As<Napi::Object>();
+      QTreeWidgetWrap* parentWidgetWrap =
+          Napi::ObjectWrap<QTreeWidgetWrap>::Unwrap(parentObject);
+      QTreeWidget* parent = parentWidgetWrap->getInternalInstance();
+      this->instance = new QTreeWidgetItem(parent);
+    } else if (constructorType == 3) {
+      Napi::Object parentObject = info[0].As<Napi::Object>();
+      QTreeWidgetWrap* parentWidgetWrap =
+          Napi::ObjectWrap<QTreeWidgetWrap>::Unwrap(parentObject);
+      QTreeWidget* parent = parentWidgetWrap->getInternalInstance();
+      Napi::Array stringsNapi = info[1].As<Napi::Array>();
+      QList<QString> list;
+      for (int i = 0; i < stringsNapi.Length(); i++) {
+        Napi::Value stringNapi = stringsNapi[i];
+        list.append(stringNapi.As<Napi::String>().Utf8Value().c_str());
+      }
+      QStringList strings = QStringList(list);
+      this->instance = new QTreeWidgetItem(parent, strings);
+    }
```

Comment:
This seems to be a bit complicated. 
Is it absoultely necessary to have all these constructor variants ? 😅 
Also, if we decide to keep it maybe we can chagne the contructorType to a enum with more descriptive namings that help us identify by just looking at it.

The side effect of this is also that the javascript version of the contructor also becomes really complex. What do you think?
My preference would be that we reduce the number of constructor variants itself.

Reply:
I'll think about it and look through and choose the constructors I want to keep. I definitely want to keep one that accepts a parent, being either QTreeWidget and QTreeWidgetItem, and the other being optional string for the text. I think that simplifies things.

Reply:
Yep makes sense 👍  Thanks for all the hard work, this is really great!